### ISSUE TITLE
Completion should not be displayed after switch expression

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#37315]
**Version Used**: 
![image](https://user-images.githubusercontent.com/41759/61412731-7acc0180-a89e-11e9-8547-42512338eec0.png)

**Steps to Reproduce**:

```C#
using System;
using System.Collections.Generic;
using System.Linq;

class C
{
    public List<int> X;
}

class D
{
    public List<int> X;
}

class E
{
    public static bool F(object obj) => obj switch
    {
        C c => c.X,
        D d => d.X,
        _ => throw null
    }|.Any(i => i == 0);
}
```

Place caret as indicated by `|`. Type `.`. This triggers completion:
![image](https://user-images.githubusercontent.com/41759/61413040-57558680-a89f-11e9-9817-bc146b603be7.png)

However, since switch expression is not primary expression dotting off of it is not valid, so no completion should be shown.

The same for this case:
![image](https://user-images.githubusercontent.com/41759/61413298-fbd7c880-a89f-11e9-8e65-638a1a59e354.png)

This one is ok:

![image](https://user-images.githubusercontent.com/41759/61413328-0d20d500-a8a0-11e9-9545-b6b41bf99c0a.png)

**Comment by CyrusNajmabadi** [comment]
Possibly the parse tree is thrown off by the `..`?
